pkg/util/kms/kms_version.go

```
@@ -107,17 +109,30 @@
 			return secrets.ErrInvalidSecretId
 		}
 		return err
 	}
 
+	if (v.k.driver.Name() == "KMIPSecret") {
+		encodedData, ok := s[v.BackendSecretName()]
+		if !ok {
+			return secrets.ErrInvalidSecretData
+		}
+		data := map[string]string{}
+		decodedString, err := base64.StdEncoding.DecodeString(encodedData.(string))
```

golangci-lint (1.64.8)

ineffectual assignment to err

(ineffassign)